### PR TITLE
Migrate /test-runs POST to /api/runs POST

### PR DESCRIPTION
**Pull request by lukebjerring**
Also improves some handler naming (too general for methods across the namespace) and fixes a bug with the populate_dev_data script (which is being deleted in https://github.com/w3c/wptdashboard/pull/248 anyway).

### CODE REVIEW

**Comment by lukebjerring** [comment]
Note that this PR leaves the existing endpoint functional, as migrating the scripts that utilize it is best done before deleting the old endpoint.

**Comment by rwaldron** [comment]
@lukebjerring quick rebase?

File: populate_dev_data_handler.go [hunk 1]
```diff
@@ -86,11 +86,14 @@ func populateDevData(w http.ResponseWriter, r *http.Request) {
 		"safari",
 	} {
 		// TODO(lukebjerring): Move wpt.fyi base URL to constant.
-		jsonURL := "https://wpt.fyi/json?platform=" + browserName
+		jsonURL := "https://wpt.fyi/results?platform=" + browserName
```
Comment (jeffcarp) [inline populate_dev_data_handler.go]: Is this diff related to this PR?

File: populate_dev_data_handler.go [hunk 2]
```diff
@@ -86,11 +86,14 @@ func populateDevData(w http.ResponseWriter, r *http.Request) {
...
 		resp, err := client.Head(jsonURL)
 
-		if urlError, ok := err.(*url.Error); !ok || urlError.Err != errUseLastResponse {
-			fmt.Fprintf(w, "Failed to fetch latest run for %s: %s\n", browserName, urlError.Error())
+		if err == nil {
+			fmt.Fprintf(w, "Got unexpected non-redirect status %d for %s\n", resp.StatusCode, jsonURL)
+			continue
+		} else if urlError, ok := err.(*url.Error); !ok || urlError.Err != errUseLastResponse {
```
Comment (jeffcarp) [inline populate_dev_data_handler.go]: The `err.(*url.Error)` construct scrambles my brain every time

File: api_handlers.go [hunk 3]
```diff
@@ -120,6 +132,56 @@ func apiTestRunHandler(w http.ResponseWriter, r *http.Request) {
...
+		return
+	}
+
+	if suppliedSecret != token.Secret {
+		http.Error(w, err.Error(), http.StatusUnauthorized)
+		return
+	}
+
+	var body []byte
+	if body, err = ioutil.ReadAll(r.Body); err != nil {
```
Comment (jeffcarp) [inline api_handlers.go]: I think you need to add `defer r.Body.Close()` somewhere in this function in order to not leak sockets.

File: test_run_handler.go [hunk 4]
```diff
@@ -16,71 +16,26 @@ package wptdashboard
...
 	"google.golang.org/appengine"
 	"google.golang.org/appengine/datastore"
 )
 
-func testRunHandler(w http.ResponseWriter, r *http.Request) {
+// testRunsHandler handles GET/POST requests to /test-runs
+func testRunsHandler(w http.ResponseWriter, r *http.Request) {
 	if r.Method == "POST" {
-		handlePost(w, r)
+		// /test-runs is the legacy POST endpoint, migrated to /api/runs and left to avoid breakages.
```
Comment (jeffcarp) [inline test_run_handler.go]: Can you make a bug and add `TODO(#n)` where `n` is the bug #?

File: api_handlers.go (hunk 3 above)
Comment (lukebjerring) [reply api_handlers.go]: Done

File: populate_dev_data_handler.go (hunk 1 above)
Comment (lukebjerring) [reply populate_dev_data_handler.go]: Not really; I'll make this tweak separately.

File: populate_dev_data_handler.go (hunk 2 above)
Comment (lukebjerring) [reply populate_dev_data_handler.go]: Yeah casting isn't so intuitive written this way.

File: test_run_handler.go (hunk 4 above)
Comment (lukebjerring) [reply test_run_handler.go]: Done.

File: api_handlers.go (hunk 3 above)
Comment (jeffcarp) [reply api_handlers.go]: Ah! TIL :+1:
